internal/openshift/client/openshift_client.go

```diff
@@ -143,6 +145,56 @@ func (o openShift) Idle(apiURL string, bearerToken string, namespace string, ser
 	return
 }
 
+// Reset deletes a pod and start a new one
+func (o openShift) Reset(apiURL string, bearerToken string, namespace string) (err error) {
+	logger.Info("reseting pods in " + namespace)
+
+	req, err := o.reqAPI(apiURL, bearerToken, "GET", namespace, "pods", nil)
+	if err != nil {
+		return err
+	}
+	resp, err := o.do(req)
+	if err != nil {
+		return err
+	}
+
+	defer bodyClose(resp)
+
+	podList := &v1.PodList{}
+	err = json.NewDecoder(resp.Body).Decode(podList)
+	if err != nil {
+		return err
+	}
+
+	for _, element := range podList.Items {
+
+		podName := element.GetName()
+		if strings.Contains(podName, "deploy") {
+			continue
+		}
+
+		log.Infof("Reseting the pod " + podName)
+		req, err := o.reqAPI(apiURL, bearerToken, "DELETE", namespace, "pods/"+podName, nil)
+		if err != nil {
+			return err
+		}
+		_, err = o.do(req)
```

Comment:
don't you have to close the body?

Reply:
body of what? Note that we are not taking any response here.

Reply:
yes, but wouldn't a response be returned anyway?

Reply:
Please see: https://golang.org/pkg/net/http/#Client.Do 

> If the returned error is nil, the Response will contain a non-nil Body **which the user is expected to close. **